### DISCUSSION TITLE
Auto Correcting Showdown Sets

### DISCUSSION THREAD

**Discussion by hexbyt3** [kwsch/PKHeX#4265] [Show and tell]
Thanks Kurt for introducing me to the Applicator's that I never knew existed this will help with the [Auto Correct Showdown Set](https://github.com/bdawg1989/MergeBot/blob/main/SysBot.Pokemon/Helpers/AutoCorrectShowdown.cs) feature i'm building into my [custom sysbot](https://github.com/bdawg1989/MergeBot).  

This really does take the most fudged up set anyone could create, correct it, and push it through!

![image](https://github.com/kwsch/PKHeX/assets/80122551/4af39e93-8c9a-427f-9361-f4561bf23ab8)

If anyone is interested and could use this for something feel free!  Also, if you have any tips or see anything I should add or change please let me know.  I feel this will help a lot of newbies with their sets.

So far these are all the settings I have for it:

![image](https://github.com/kwsch/PKHeX/assets/80122551/36a8cd81-fb26-42a1-962c-78d08b7655a6)

I think the next steps I'll take is working on a way to correct batch commands like Met Locations, Ribbons, Marks, etc. and utilize the new NicknameVerifier  
I also want to refactor the level adjustment logic to adjust the level based on moves, legal ball, and evolution requirements, but that sounds like a bit of a pain.  Thoughts?

**Comment by kwsch** [comment]
It's a trade-off when accepting improperly formatted user inputs. I personally would just reject the set and indicate the reasons, because stuff like EV oopsies can be typos that result in an incorrect spread. Rejecting lets the user correct the error and avoids a situation where you assumed one way when the user intended another.
Ideally the user uses a set-builder like Showdown rather than manually fudge things, but alas...

**Reply by hexbyt3** [reply]
That's a good point.  I eventually want to add in good reasons like the other bot I built has if the bot owner opts out of auto correcting the showdown set.  My "GrandOak" bot is at it's 100 server limit, and it's been rather hard getting Discord to verify it.  So I had to eventually go this route and add it to the sysbot code.

This is what my python bot does currently:

![image](https://github.com/kwsch/PKHeX/assets/80122551/caa3b2a5-a50e-4b25-867d-5ad2f6bc4db2)


Which is my goal to add to the AutoCorrectShowdown code as well.
